src/Infrastructure.Dapper/Repositories/CollectionRepository.cs

```diff
@@ -82,6 +95,87 @@ public async Task<ICollection<Collection>> GetManyByOrganizationIdAsync(Guid org
         }
     }
 
+    public async Task<ICollection<Tuple<Collection, CollectionAccessDetails>>> GetManyByOrganizationIdWithAccessAsync(Guid organizationId)
+    {
+        using (var connection = new SqlConnection(ConnectionString))
+        {
+            var results = await connection.QueryMultipleAsync(
+                $"[{Schema}].[Collection_ReadWithGroupsAndUsersByOrganizationId]",
+                new { OrganizationId = organizationId },
+                commandType: CommandType.StoredProcedure);
+
+            var collections = (await results.ReadAsync<Collection>()).ToList();
+            var groups = (await results.ReadAsync<CollectionGroup>()).ToList();
+            var users = (await results.ReadAsync<CollectionUser>()).ToList();
```

Comment:
question: Do we really need to convert theses to lists? Shouldn't it be enough if we convert the query results below to lists? I think I saw a discussion on this somewhere but I'm not sure where it ended up.

Reply:
Comment is valid for rows 147-139 further down in this file